students/tammyd/lesson09/mailroom_oo.py

```diff
@@ -0,0 +1,235 @@
+    @property
+    def total_donations(self):
+        return sum(self._donations)
+
+    @property
+    def new_donation(self, donation):
+        self._donations.append(donation)
+
+    @property
+    # total_donations = sum(values)
+    def total_donations(self):
+        return sum(self._donations)
+
+    @property
+    # num_donations = len(values)
+    def num_donations(self):
+        return len(self._donations)
+
+    @property
+    # avg_donations = total_donations / num_donations
+    def avg_donations(self):
+        return sum(self._donations) / len(self._donations)
+
+
+
+
+#You’ll then want a class that handles the collection of donors. 
+#method needs the information from the whole collection.
+#pt1: data structure that holds a list of your donors
+#Do I want to include or exclude () right after DonorDB here?
+class DonorDB:
+    #donor objects
+    def __init__(self):
+        self._donors = {}
+
+
+    #methods to add a new donor
+    def add_donor(self, donor):
+        self._donors[donor.name.lower()] = donor
+    #search for a given donors
+    #save and re-load your data
+
+#pt1: Send a Thank You
+def thank_you():
+    print("\nMenu: Send a Thank You\n")
+    thank_you_dict = {"list": get_donor_database,
+                      "menu": main_menu,
+                      "q": quit_program
+                      }
+    while True:
+        donor_name = input("Enter the donor's full name. \n-- or 'list' to see a list of donors, \n-- or 'menu' to return to the main menu: ").strip()
+        donor_name = donor_name.lower()
+        try: 
+            return thank_you_dict[donor_name]()
+
+        # if donor_name == "list":
+        #     print("\nDonor database:")
+        #     for keys in donor_db:
+        #         print(keys.title())
+        #     print("\n")
+        # elif donor_name == "menu":
+        #     return main_menu()
+        # elif donor_name == "q":
+        #     return quit_program()
+        # else:
+        #     break
+        except KeyError:
+            print("Incorrect entry. Try again.")
+            return
+    donation(donor_name)
+
+
+
+def find_donor(donor_name):
+    for donor in donor_db:
+        if donor_name.strip().lower() == donor[0].lower():
+            return donor
+    return None
+
+def ty_letter(donor_name, donor_amount):
+    print("\nMenu: Send a Thank You, Letter\n")
+    letter = (f"\nDear {donor_name.title()},\n\n\tYour donation of ${donor_amount} will make a great difference to a young learner at the coding academy. Thank you.\n\n\nSincerely,\nAn Academy Student\n")
+    print(letter)
+    print("--------------------------------\n\n")
+    main_menu()
+
+
+def letters_to_everyone():
+    for donor_name in donor_db.values():
+        letter = (f"\nDear {donor_name[0]},\n\n\tYour donation of ${donor_name[1][-1]} will make a great difference to a young learner at the coding academy. Thank you.\n\n\nSincerely,\nAn Academy Student\n")
```

Comment:
this is the same code as the previous function! -- you should call that from this function.